### PR TITLE
Issue #16139 - Fix: Add check to detect whether cmake exists

### PR DESCRIPTION
**Pull request by omkaark**
Issue https://github.com/yugabyte/yugabyte-db/issues/16139, added check to only execute `which cmake` if cmake exists. Otherwise, print error and exit with Code 1.

### CODE REVIEW

**Comment by omkaark** [comment]
Any clue why deploy failed? https://app.netlify.com/sites/infallible-bardeen-164bc9/deploys/643ee7d4566c6300080aaf5e

File: yb_build.sh [hunk 1]
```diff
@@ -461,6 +461,10 @@ run_cxx_build() {
     if is_mac && [[ "${YB_TARGET_ARCH:-}" == "arm64" ]]; then
       cmake_binary=/opt/homebrew/bin/cmake
     else
+      if ! which cmake &> /dev/null; then
+          echo "Error: cmake not found in PATH" >&2
+          exit 1
```
Comment (jasonyb) [inline yb_build.sh:466]:
```suggestion
        echo "Error: cmake not found in PATH" >&2
        exit 1
```

**Review by jasonyb — APPROVED** [review]
Thanks.

File: yb_build.sh (hunk 1 above)
Comment (omkaark) [reply yb_build.sh]: No worries, looking forward to making more contributions! Love all the value yugabyte provides :)

**Pull request merged**